app/src/main/java/com/codesoom/assignment/TaskController.java

```diff
@@ -0,0 +1,155 @@
+package com.codesoom.assignment;
+
+import com.codesoom.assignment.enums.HttpMethodType;
+import com.codesoom.assignment.error.ClientError;
+import com.codesoom.assignment.models.Path;
+import com.codesoom.assignment.models.RequestTaskDTO;
+import com.codesoom.assignment.models.Task;
+import com.codesoom.assignment.service.TaskService;
+import com.codesoom.assignment.utils.JsonUtil;
+import com.codesoom.assignment.utils.StringUtil;
+import com.sun.net.httpserver.HttpExchange;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Optional;
+
+public class TaskController {
+    private static final TaskController taskController = new TaskController();
+    private final TaskService taskService = TaskService.getInstance();
+
+    private TaskController() {
+    }
+
+    public static TaskController getInstance() {
+        return taskController;
+    }
+
+    public void requestMapping(HttpExchange exchange,
+                               HttpMethodType method,
+                               Path path,
+                               String requestBody) throws IOException {
+        if (isRequiredPathVariable(method) && path.getPathVariable() == null) {
+            ClientError.badRequest(exchange);
+            return;
+        }
+
+        if (isRequiredRequestBody(method) && requestBody.isBlank()) {
+            ClientError.badRequest(exchange);
+            return;
+        }
+
+        switch (method) {
+            case GET:
+                if (path.getPathVariable() == null) {
```

Comment:
`null`을 쓰지 않고 이 `if`문이 일상어처럼 읽히도록 코드를 수정해 보세요.
예를 들어 이렇게 할 수 있겠죠.

```suggestion
                if (path.isInvalid()) {
```

Reply:
비교식도 일상어처럼 읽히도록 메서드로 추출하는 것을 습관들여야겠네요 :)
pathVariable이 비어있는지 확인하는 메서드로써, isEmptyPathVariable(path)를 사용해보았습니다!
```java
    if (isEmptyPathVariable(path)) {
        gets(exchange);
        return;
    }

    private boolean isEmptyPathVariable(Path path) {
        return path.getPathVariable() == null;
    }
```